tests/keras_tests/function_tests/test_hessian_service.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+
+import keras
+import numpy as np
+from tensorflow import initializers
+from tensorflow.keras.layers import Conv2D, BatchNormalization, ReLU, Input
+
+import model_compression_toolkit.core.common.hessian as hess
+from model_compression_toolkit.core.common.hessian import TraceHessianService, TraceHessianConfig, TraceHessianRequest, TraceHessianMode, \
+    TraceHessianGranularity
+from model_compression_toolkit.core.keras.default_framework_info import DEFAULT_KERAS_INFO
+from model_compression_toolkit.core.keras.keras_implementation import KerasImplementation
+from model_compression_toolkit.target_platform_capabilities.tpc_models.imx500_tpc.latest import generate_keras_tpc
+from tests.common_tests.helpers.prep_graph_for_func_test import prepare_graph_with_configs
+
+
+def basic_model(input_shape):
+    random_uniform = initializers.random_uniform(0, 1)
+    inputs = Input(shape=input_shape)
+    x = Conv2D(2, 3, padding='same', name="conv2d")(inputs)
+    x_bn = BatchNormalization(gamma_initializer='random_normal', beta_initializer='random_normal',
+                              moving_mean_initializer='random_normal', moving_variance_initializer=random_uniform,
+                              name="bn1")(x)
+    outputs = ReLU()(x_bn)
+    return keras.Model(inputs=inputs, outputs=outputs)
+
+
+def representative_dataset(num_of_inputs=1):
+    yield [np.random.randn(2, 8, 8, 3).astype(np.float32)] * num_of_inputs
+
+
+class TestHessianService(unittest.TestCase):
```

Comment:
Just verifying - did you include a test that asks for more images than what is already computed and triggers a computation after fetching the cached results? If not, I think this is an important one.